### ISSUE TITLE
[Feature request] Hide selenium use in the drivers

### ISSUE THREAD

**Issue by GoostDon** [jjjchens235/bing-rewards#208]
So I saw the discussion about random amount of searches. I don't feel much for this feature but you(jjjchens235) mentioned something about MS detecting selenium browsers. I was surprised that it is possible and yet MS is not doing it (that we know of). I truly believe they will do it in the future.

So I think it will be a good feature to hide selenium for MS.

See the following solution: https://stackoverflow.com/a/52108199

P.S I am new to github so I am not sure if I am posting this feature request at the right place

Edit: typos(no native English speaker)

**Comment by GoostDon** [comment]
@DimaRazb I am sorry to hear that. Are you sure it is because of selenium and not because of something else? Did MS give you a reason for your bans? If the bans are truly because of the use of selenium, this feature may need to have a high prio status.

**Comment by DimaRazb** [comment]
Nah, I didn't contact the support.

**Comment by jjjchens235** [comment]
The linked solution is to open up `chromedriver` file in a hex editor or vim and then replace the `cdc_` string.

However, I'm not sure how this would be done programmatically, I tried something like this:
https://stackoverflow.com/a/17141572

But got this error `UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte`

**Comment by jhaugh0** [comment]
> The linked solution is to open up `chromedriver` file in a hex editor or vim and then replace the `cdc_` string.
> 
> However, I'm not sure how this would be done programmatically, I tried something like this: https://stackoverflow.com/a/17141572
> 
> But got this error `UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte`

Perl works really well for this. I'm currently using a powershell wrapper around the whole thing for reasons, but in powershell I solved it by this:

```
if (cat $directory/bing/bing-rewards/BingRewards/drivers/chromedriver | grep -a "cdc_"){
        $replacement = "$(get-random ('a'..'z'))$(get-random ('a'..'z'))$(get-random ('a'..'z'))_"
        Write-Host "<$(get-date -Format HH:mm:ss)> cdc_ detected in chromedriver, replacing with $replacement"
        perl -pi -e "s/cdc_/$replacement/g" $directory/bing/bing-rewards/BingRewards/drivers/chromedriver
}
```

Still in the process of learning python, but it looks like you could drop the perl command into a standalone file, then call that

https://stackoverflow.com/questions/4682088/call-perl-script-from-python

**Comment by jjjchens235** [comment]
Makes sense. I have no knowledge of Perl. Maybe you can create a PR for the Perl portion of this, and I can invoke it with python?

It looks like you're currently using a mixture of shell and Perl commands, would it be possible to put all the logic in the Perl file?

**Comment by Shymaxtic** [comment]
Hi all, I have same opinion that MS is checking the tool using webdriver. Recently my account is blocked or declined when redeeming gift card. I followed https://stackoverflow.com/a/17141572 but nothing better.
Is there anyone successfully get the gift card by using this tool?

**Comment by GoostDon** [comment]
So I did some more research and read this article: https://piprogramming.org/articles/How-to-make-Selenium-undetectable-and-stealth--7-Ways-to-hide-your-Bot-Automation-from-Detection-0000000017.html

The article links to https://github.com/ultrafunkamsterdam/undetected-chromedriver in point 11.
This is (if I understand it correctly) a project that makes the chromedriver undectable(Passes ALL bot mitigation systems). It frequently provides updates so maybe we can make use of this driver instead of the default chromedriver.

I have not found a project that does the same with the edge browser (yet). But I love the idea to start with the chromedriver and 
and learn before improving the edge browser.

The other points mentioned in the article, were also intressting. Especially point 3. I haven't had time yet to look in the code for possible improvements about this subject. Anyway, I wanted to share this information. Maybe someone smarter and faster than me, already knows the best fixes for this issue.

I will provide sample codes once I made adjustments and found it working. But don't aspect that anywhere soon since i just became a father.
